fitwsarah_frontend/src/components/clientProfile/Appointment.js

```diff
@@ -31,14 +49,75 @@ function Appointment( {appointment, accessToken} ) {
             return response.json();
         })
         .then((data) => {
-            console.log(data);
             setService(data);
         })
         .catch((error) => {
             console.error("Error message:", error.message);
         });
     };
 
+    function cancelAppointment(appointmentId) {
+        fetch(`${process.env.REACT_APP_BASE_URL}/api/v1/appointments/${appointmentId}/cancelled`, {
+            method: "PUT",
+            headers: {
+                Authorization: "Bearer " + accessToken,
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify("CANCELLED")
+        })
+        .then((response) => {
+            if (!response.ok) {
+                console.error("Response status: " + response.status);
+                throw new Error('Network response was not ok ' + response.statusText);
+            }
+            console.log("Appointment cancelled");
+        })
+        .catch((error) => {
+            console.error("Error message:", error.message);
+        });
+    };
+
+    const date_diff_indays = function (date1, date2) {
+        const dt1 = new Date(date1);
+        const dt2 = new Date(date2);
+        return Math.floor(
+            (Date.UTC(dt2.getFullYear(), dt2.getMonth(), dt2.getDate()) -
+                Date.UTC(dt1.getFullYear(), dt1.getMonth(), dt1.getDate())) /
+                    (1000 * 60 * 60 * 24)
+        );
+    };
```

Comment:
What does this function do?

Reply:
It calculate the number of days between the appointment date and todays date